config/crds/kudo.dev_instances.yaml

```diff
@@ -3,44 +3,161 @@
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.4
```

Comment:
do we have to have this annotation as it will create diff on CRDs every time we update?

Reply:
I'm assuming by "every time we update" you mean bumping the controller-gen dependency. Do you think we'll be doing this more often than every few months?

It's being (added unconditionally](https://github.com/kubernetes-sigs/controller-tools/blob/a5fa7b956b85a6e792bc7086fedf7107d62452b1/pkg/crd/gen.go#L110), so we have the following options:
1. leave it as is - after bumping the dependency we can update this everywhere automatically with `make generate && grc -c conf.gotest go test -tags integration  ./pkg/kudoctl/cmd -v -mod=readonly --update -run TestInitCmd_yamlOutput`
1. If you think this is still too much of a hassle, add a post-processing step to `make generate` which will delete this annotation with something like https://github.com/mikefarah/yq

Reply:
it's not big deal, but we recently removed annotation of `controller-tools: 1.0` from our CRDs which was something similar in my head so I was just wondering if we need this one. I don't think it's worth the hassle of removing it though :) Thanks for listing the options though